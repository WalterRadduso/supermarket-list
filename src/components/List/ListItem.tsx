import React from "react";

import Spinner from "../Spinner";
import deleteIcon from "../../assets/delete-icon.svg";

import styles from "./ListItem.module.scss";

interface Props {
  deletingItem?: boolean;
  onRemove: VoidFunction;
}

const ListItem: React.FC<Props> = ({children, deletingItem = false, onRemove}) => {
  return (
    <li className={styles.container}>
      <span>{children}</span>

      <button onClick={onRemove}>
        {deletingItem ? <Spinner /> : <img alt="Delete item image" src={deleteIcon} />}
      </button>
    </li>
  );
};

export default ListItem;
