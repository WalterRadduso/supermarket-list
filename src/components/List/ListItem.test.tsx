import React from "react";
import {render, screen, fireEvent} from "@testing-library/react";
import {renderHook} from "@testing-library/react-hooks";

import ListItem from "./ListItem";

describe("ListItem Tests", () => {
  const onRemove = jest.fn();

  test("should render snapshot", () => {
    const listItemRendered = render(<ListItem onRemove={onRemove}>List items</ListItem>);

    expect(listItemRendered).toMatchSnapshot();
  });

  test("should render snapshot with delete", () => {
    const listItemRendered = render(
      <ListItem deletingItem={true} onRemove={onRemove}>
        List items deleting
      </ListItem>,
    );

    expect(listItemRendered).toMatchSnapshot();
  });

  // test("should render list with scroll", () => {
  //   const setScrollableList = jest.fn();
  //   const ref = {
  //     current: {
  //       scrollHeight: 1000,
  //       scrollTo: jest.fn(),
  //     },
  //   };
  //
  //   renderHook(() =>
  //     useScrollTo({itemsLength: 20, listRefCurrent: ref.current, setScrollableList}),
  //   );
  //
  //   expect(ref.current.scrollTo).toHaveBeenCalledTimes(1);
  // });
  //
  // test("should render list without scroll", () => {
  //   const setScrollableList = jest.fn();
  //   const ref = {
  //     current: {
  //       scrollHeight: 400,
  //       scrollTo: jest.fn(),
  //     },
  //   };
  //
  //   renderHook(() =>
  //     useScrollTo({itemsLength: 20, listRefCurrent: ref.current, setScrollableList}),
  //   );
  //
  //   expect(setScrollableList).toHaveBeenCalledTimes(1);
  // });
});
