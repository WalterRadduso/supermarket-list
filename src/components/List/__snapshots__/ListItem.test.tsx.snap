// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem Tests should render snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="container"
      >
        <span>
          List items
        </span>
        <button>
          <img
            alt="Delete item image"
            src="delete-icon.svg"
          />
        </button>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="container"
    >
      <span>
        List items
      </span>
      <button>
        <img
          alt="Delete item image"
          src="delete-icon.svg"
        />
      </button>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ListItem Tests should render snapshot with delete 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="container"
      >
        <span>
          List items deleting
        </span>
        <button>
          <div
            class="container"
          >
            <div
              class="spinnerContainer"
            >
              <span
                class="spinner secondary"
              />
            </div>
          </div>
        </button>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="container"
    >
      <span>
        List items deleting
      </span>
      <button>
        <div
          class="container"
        >
          <div
            class="spinnerContainer"
          >
            <span
              class="spinner secondary"
            />
          </div>
        </div>
      </button>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
